### DISCUSSION TITLE
RAG - Someone can give me an help ?

### DISCUSSION THREAD

**Discussion by peuportier** [open-webui/open-webui#4493] [Q&A]
Dear Community,

I am exploring the possibility of enhancing our RAG (Retrieval-Augmented Generation) system by integrating the Longformer model to better handle and process large documents like reports. The Longformer is designed to efficiently manage longer texts by utilizing a unique attention mechanism, which could significantly improve our system's ability to parse and understand extensive data sets.

Could anyone share insights or experiences regarding the integration of the Longformer model into a RAG system? Specifically, I am interested in:

Which specific Longformer model has been successfully integrated with RAG systems before? For example, are there specific pre-trained versions or configurations of Longformer that are recommended?
What would be the optimal chunk size and overlap when using Longformer in this context? Understanding the best parameters for chunk size and overlap can help in fine-tuning the model to maximize efficiency and effectiveness.
Any advice or suggestions on this integration would be greatly appreciated, including any potential challenges and how they might be addressed.

Thank you!

**Comment by peuportier** [comment]
Anyone please. Just please let me know how do I add into openwebui a longformer model. Thanks you

**Comment by PlebeiusGaragicus** [comment]
See my [earlier post](https://github.com/open-webui/open-webui/discussions/4823#discussioncomment-10457731) on current OI RAG

I'm not sure what you're looking for is a specific model to use for rag or specific hyper parameter tunings for your chunk size, etc.

I think a more mature RAG solution is needed - not just a single pass of query -> vector store retrieval -> LLM

**Comment by peuportier** [comment]
Dear @PlebeiusGaragicus 

I sincerely thank you for taking the time to respond to our inquiry. To introduce ourselves, we are a research lab based in France and affiliated with the Pasteur Institute.

We are currently facing challenges with input/output (IO) related to processing very long-form documents. Specifically, we are looking for a more suitable model for this task. Would you recommend using a specialized model such as a Longformer or T5? Additionally, how would one go about implementing these models in the IO process? Alternatively, should we focus on adjusting parameters like chunk size to achieve better results?

For instance, when we input a PDF containing a table with 300 entries and ask the model to list and sum them, it only processes a portion of the document and does so inconsistently. This approach is not practical for our needs.

We would greatly appreciate any insights or advice you could provide. Thank you again for your time. KR Aurelien

**Reply by peuportier** [reply]
We truly appreciate using IO, as it has proven to be an excellent tool overall. It's especially valuable when used on-premise in a local environment, which is crucial for keeping our research data within a controlled setting. While we can utilize services like ChatGPT or other online AI providers, the reliability and performance of IO stand out.

I admire the team's efforts to continually enhance this tool, making it increasingly powerful. However, there's one aspect where we would appreciate more efficiency, or perhaps guidance—specifically, how to effectively use IO for processing longer documents.
